tests/discover/filtering.sh

```diff
@@ -45,6 +45,170 @@ rlJournalStart
+        rlRun "tmt run -dvvvr --all $plan discover -h fmf --fmf-id | tail -n1 \
+               | tee output"
+        rlAssertGrep "path:" output
+    rlPhaseEnd
+
+    # If plan or test weren't explicitly specified then fmf-ids for all tests
+    # in all plans should be shown
+    rlPhaseStartTest "fmf-id (w/o url): plans were executed if plan/test -n=."
+        ids_amount=$(tmt run -r discover -h fmf --fmf-id finish |
+                     grep "name:" |
+                     wc -l)
+        tests_amount=$(tmt run -r discover -h fmf finish |
+                       grep "summary:" |
+                       awk '{print $2}' |
+                       awk '{s+=$1} END {print s}')
+        rlAssertEquals "Check that number of fmf-ids equals to tests number" \
+                       "$ids_amount" "$tests_amount"
+    rlPhaseEnd
+
+    # Checking the case when fmf_root = git_root
+    rlPhaseStartTest "fmf-id (w/o url): path doesn't shown up"
+        path="$(git rev-parse --show-toplevel)"
+        rlRun "cd $path"
+        rlRun "tmt run -r test --name /tests/unit \
+                          plans --default \
+                          discover --how fmf --fmf-id finish | tee output"
+        rlAssertNotGrep "path:" output
+    rlPhaseEnd
+
+#    # If the test exists in 2 or more plans than the test should be printed
+#    # only once
+#    rlPhaseStartTest "fmf-id: one test exists in 2 or more plans"
+#        path="$(git rev-parse --show-toplevel)"
+#        rlRun "cd $path"
+#        ids_amount=$(tmt run -r discover --how fmf --fmf-id \
+#                     tests --name /tests/unit |
+#                     grep "name:" |
+#                     wc -l)
+#        tests_amount=1
+#        rlAssertEquals "Check that number of fmf-ids equals to tests number" \
+#                       "$ids_amount" "$tests_amount"
+#    rlPhaseEnd
+
+    rlPhaseStartTest "fmf-id (w/o url): check the test with --how=shell"
+        path="$(git rev-parse --show-toplevel)"
+        rlRun "cd $path/plans/sanity"
+        rlRun "tmt run -r test --name /lint/plans \
+                          plan --name /plans/sanity/lint \
+                          discover -h shell --fmf-id finish 2>&1 | \
+                          tee output" 2
+        rlAssertGrep "Error: no such option: --fmf-id" output -i
+    rlPhaseEnd
+
+    # Raise an exception if --fmf-id uses w/o --url and git root doesn't exist
+    rlPhaseStartTest "fmf-id (w/o url): and git root doesn't exist"
+        tmp_dir="$(mktemp -d)"
+        rlRun "cd $tmp_dir"
+        rlRun "tmt init --template base"
+        rlRun "tmt run -rdvvv discover -h fmf --fmf-id finish 2>&1 \
+               | tee output" 2
+        rlAssertGrep "\`tmt run discover --fmf-id\` without \`url\` option \
+can be used only within git repo." output
+        rlRun "rm -rf $tmp_dir"
+    rlPhaseEnd
+
+    # If in non-git directory exist plan w/ and w/o url, then only fmf-ids for
+    # the plan w/ url should be reported
```

Comment:
Is this assumption correct? If I have one plan which cannot report fmf-id if called in its own, why this error should be hidden if it is called with another but working plan? --fmf-id is expected for all plans not only those which can do it.

Reply:
Thanks for the review @lukaszachy, in this case,
1) If we report an error
`tmt run discover --fmf-id without url option can be used only within git repo.` it will be wrong for "working plan" as it has url. 
2) If we report an error then we shouldn't show fmf-ids for "working plan" also. Otherwise we will have a case:
```
(venv) [dkarpele@redhat plans]$ ls
example.fmf  non-url.fmf
(venv) [dkarpele@redhat plans]$tmt run -rv discover -h fmf --fmf-id finish
---
name: /tests/clean/guests
url: https://github.com/teemtee/tmt
ref: main
...
`tmt run discover --fmf-id` without `url` option can be used only within git repo.
```

It breaks the original idea of beaker workflow. `tmt run plan --fmf-id | bkr workflow-tomorrow f-34 x86_64 --fmf-id -`. We can't pass error message to bkr.
OR if non url file goes alphabetically first.
```
(venv) [dkarpele@redhat plans]$ ls
a-non-url.fmf  example.fmf
(venv) [dkarpele@redhat plans]$ tmt run -rdddvvv discover -h fmf --fmf-id finish
`tmt run discover --fmf-id` without `url` option can be used only within git repo.
```
It breaks the original idea of beaker workflow. `tmt run plan --fmf-id | bkr workflow-tomorrow f-34 x86_64 --fmf-id -`. We can't pass error message to bkr.